community/kernel/src/main/java/org/neo4j/kernel/impl/store/ShortArray.java

```diff
@@ -782,17 +783,15 @@ public static ShortArray typeOf( Object array )
 
     public static int calculateNumberOfBlocksUsed( long firstBlock )
     {
-        Bits bits = Bits.bitsFromLongs( new long[] {firstBlock} );
-        // bbbb][bbll,llll][yyyy,tttt][kkkk,kkkk][kkkk,kkkk][kkkk,kkkk]
-        bits.getInt( 24 ); // Get rid of key
-        bits.getByte( 4 ); // Get rid of short array type
-        bits.getByte( 4 ); // Get rid of the type
-        int arrayLength = bits.getByte( 6 );
-        int requiredBits = bits.getByte( 6 );
+        int highInt = (int) (firstBlock >>> 32);
+        int arrayLength = highInt & 0b11_1111;
+        highInt >>>= 6;
+        int requiredBits = highInt & 0b11_1111;
```

Comment:
Please add a comment detailing what this is doing, and its approach to doing it.